File: problems/algorithms/function-calendar/php/response/src/calendar.php
<?php

/** calendar
 * $beginWeek: 0..6 - 0(DOM), 1(SEG), 2(TER), 3(QUA), 4(QUI), 5(SEX), 6(SAB)
 * $endDay: 28..31
 */
function calendar($beginWeek, $endDay)
{
  $result = "DOM SEG TER QUA QUI SEX SAB\n";

  for ($times = 0; $times < $beginWeek; $times += 1) {
    $result .= '    ';
  }

  for ($day = 1; $day  <=  $endDay; $day += 1) {
    $result .= ($day < 10) ? " 0${day}" : " ${day}";
    if ($day == $endDay)
      return  $result;
    $result .= ($day + $beginWeek) % 7 == 0 ? "\n" : ' ';
  }
}

File: problems/algorithms/function-calendar/ruby/response/src/calendar.rb
def calendar(start, days)
  result = "DOM SEG TER QUA QUI SEX SAB\n"
  v = 0
  while(v < start)
    result << "    "
    v += 1
  end
  d = 1
  while(d <= days)
    d < 10 ? result << " 0#{d} " : result << " #{d} "
    if (d == days)
      return result
    end
    (d + start) % 7 == 0 ? result << "\n" : " "
    d += 1
  end
  return result
end